From c4cfd0a5b13cb3a03f00ce1fb8855e17d0ed44ff Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 17 Dec 2024 11:31:22 +0100
Subject: [PATCH] ARM: dts: stm32: support of continuous display on stm32mp157d
 EV1 board

Add DSI clocks to simple framebuffer device & set default-on
property to ltdc & dsi.

Change-Id: I8c11a0011f73665ed5559362cfcb0eb5da34a6f0
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/425201
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/st/stm32mp157d-ev1.dts | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
index 22410107f594..4631832ee824 100644
--- a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
@@ -29,7 +29,7 @@ chosen {
 
 		framebuffer {
 			compatible = "simple-framebuffer";
-			clocks = <&rcc LTDC_PX>;
+			clocks = <&rcc LTDC_PX>, <&rcc DSI>, <&rcc DSI_PX>;
 			status = "disabled";
 		};
 	};
@@ -134,6 +134,7 @@ panel_backlight: panel-backlight {
 		compatible = "gpio-backlight";
 		gpios = <&gpiod 13 GPIO_ACTIVE_LOW>;
 		default-on;
+		default-brightness-level = <1>;
 		status = "okay";
 	};
 
@@ -339,6 +340,7 @@ dfsdm_endpoint3: endpoint {
 &dsi {
 	#address-cells = <1>;
 	#size-cells = <0>;
+	default-on;
 	status = "okay";
 
 	panel_dsi: panel@0 {
@@ -349,6 +351,7 @@ panel_dsi: panel@0 {
 		power-supply = <&v3v3>;
 		vcc-supply = <&v3v3>;
 		iovcc-supply = <&v3v3>;
+		default-on;
 		status = "okay";
 
 		port {
@@ -547,6 +550,7 @@ &i2c5 {
 };
 
 &ltdc {
+	default-on;
 	status = "okay";
 
 	port {
-- 
2.50.1

